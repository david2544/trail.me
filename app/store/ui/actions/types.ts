import { ActionType } from 'typesafe-actions';
import * as actions from '@store/ui/actions';

/* --- STATE --- */
interface UIState {
  readonly locale: string;
}

/* --- ACTIONS --- */
type UIActions = ActionType<typeof actions>;

/* --- EXPORTS --- */
export { UIActions, UIState };
