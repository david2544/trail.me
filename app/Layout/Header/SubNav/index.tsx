import React from 'react';
import { Link } from 'react-router-dom';
import classnames from 'classnames';
import styles from './styles.module.scss';

interface SubNavProps {
  subMenuItem: {
    link: string;
    title?: string;
    titleSub?: string;
    icon?: string;
    description: string;
    isExternal?: boolean;
  };
}

const SubNav: React.FC<SubNavProps> = ({ subMenuItem }) =>
  subMenuItem.isExternal ? (
    <li className={styles.subNavExternal}>
      <div className={styles.subNavItem}>
        <a
          href={subMenuItem.link}
          className={styles.link}
          target="_blank"
          rel="noopener noreferrer"
        >
          <div className={styles.menuTitle}>
            <div className={styles.menuTitleMain}>{subMenuItem.title}</div>
            <div className={styles.menuTitleSub}>{subMenuItem.titleSub}</div>
          </div>
          <div className={styles.menuDescription}>{subMenuItem.description}</div>
        </a>
      </div>
    </li>
  ) : (
    <li className={styles.subNavItem}>
      <Link to={subMenuItem.link} className={styles.link}>
        <div className={styles.menuTitle}>{subMenuItem.title}</div>
        <div
          className={classnames(styles.menuDescription, {
            [styles.menuDescriptionSpacing]: !!subMenuItem.title,
          })}
        >
          {subMenuItem.description}
        </div>
      </Link>
    </li>
  );

export default SubNav;
