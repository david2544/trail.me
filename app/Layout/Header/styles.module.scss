@import '../Styles/styles.scss';

.mainHeaderNav {
  position: absolute;
  z-index: 1001;
  width: 100%;
  top: 0;
  color: white;

  .logoContainer {
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 2;
    padding-top: env(safe-area-inset-top);
    transition: margin-top 100ms ease-in-out;

    @media (min-width: 768px) {
      height: 10px;
      background: white;
      box-shadow: 2px 3px 6px 0 rgba(0, 0, 0, 0.25);
    }
  }

  .logoWrapper {
    position: relative;
    transition: transform 100ms ease-in-out;
    transform: scale(1, 1);
    transform-origin: top left;
    display: inline-block;

    @media (max-width: 767px) {
      position: relative;
    }

    &:before {
      content: '';
      display: block;
      height: 10px;
      width: 100%;
      background: white;
      position: relative;
      z-index: 1;
    }

    .logo {
      display: flex;
      background: white;
      box-shadow: 2px 3px 6px 0 rgba(0, 0, 0, 0.25);
      border-bottom-left-radius: 4px;
      border-bottom-right-radius: 4px;
      padding: 5px 10px 15px;

      @media (max-width: 1199px) {
        height: 60px;
      }

      @media (max-width: 991px) {
        height: 50px;
      }
    }
  }

  @media (min-width: 768px) {
    .minimized {
      transform: scale(0.7, 0.7);
    }
  }

  .navContainer {
    @media (min-width: 992px) {
      width: 970px;
      pointer-events: none;

      a,
      .darkModeToggle {
        pointer-events: all;
      }
    }

    @media (min-width: 768px) and (max-width: 991px) {
      width: 750px;
      pointer-events: none;

      a,
      .darkModeToggle {
        pointer-events: all;
      }
    }
  }

  .burgerMenu {
    background: none;
    color: inherit;
    border: none;
    outline: inherit;
    display: none;
    padding: 10px;
    margin-top: 17px;
    float: right;
    cursor: pointer;

    span {
      display: block;
      background: $primary;
      height: 2px;
      width: 25px;
      margin-bottom: 5px;
      transition: all 200ms ease-in-out;
      position: relative;
      top: 0;
      bottom: 0;
      transform: rotate(0deg);

      &:last-child {
        margin-bottom: 0;
      }
    }
  }

  .menuWrapper {
    padding: 0;
    list-style-type: none;
    margin: 10px 0 9px 25%;
    display: flex;

    .menuItem {
      background: none;
      border: none;
      outline: inherit;
      padding: 20px 5px;
      display: inline-block;
      color: inherit;
      position: relative;
      z-index: 2;

      &:hover,
      &:focus {
        text-decoration: underline;
      }
    }

    .menuItemWrapper {
      display: inline-block;
      flex: 1;
      text-align: center;
      position: relative;

      @media (min-width: 768px) {
        &.active {
          button {
            position: relative;

            &:after {
              content: '';
              display: block;
              width: 3px;
              height: 3px;
              background-color: white;
              border-radius: 3px;
              margin-top: 9px;
              margin-left: -1px;
              pointer-events: none;
              position: absolute;
              z-index: -1;
              left: 50%;
            }
          }
        }

        .subNav {
          pointer-events: none;
          position: absolute;
          color: black;
          margin: -26px 0 0 -18px - 30px;
          padding: 8px 0 0 0;
          font-size: 14px;
          left: 50%;
          text-align: left;
          white-space: nowrap;
          transition: all 200ms ease-in-out;
          transform: scale(0.01, 0.01);
          transform-origin: 18px + 30px 0%;
          opacity: 0;

          ul {
            list-style-type: none;
            margin: 0;
            box-shadow: 0 0 7px 0 rgba(0, 0, 0, 0.25);
            padding: 5px 18px;
            background: white;
            margin-top: 7px;
            border-radius: 4px;
          }

          &:before,
          &:after {
            content: '';
            display: block;
            border-top-left-radius: 4px;
            width: 15px;
            height: 15px;
            background-color: white;
            transform: rotate(45deg);
            position: absolute;
            top: 8px;
            left: 39px;
            z-index: 2;
            pointer-events: none;
          }

          &:after {
            z-index: -1;
            box-shadow: 0 0 7px 0 rgba(0, 0, 0, 0.25);
          }

          li {
            a,
            .darkModeToggle {
              border-bottom: 1px solid rgba(0, 0, 0, 0.1);
              padding: 12px 0;
              position: relative;
              display: block;
              text-decoration: none;
            }

            &.active a,
            a.active {
              font-weight: bold;
            }

            a:hover,
            a:focus {
              text-decoration: underline;
            }

            &:last-child {
              a {
                border-bottom: 0;
              }
            }
          }
        }

        button:hover + .subNav,
        button:focus + .subNav,
        a:hover + .subNav,
        a:focus + .subNav,
        .subNav:hover {
          pointer-events: auto;
          transform: scale(1, 1);
          opacity: 1;
          z-index: 3;
        }

        button:not(:hover) + .subNav:not(:hover),
        a:not(:hover) + .subNav:not(:hover) {
          transition: all 200ms ease-in-out;
          transform: scale(0.01, 0.01);
          transform-origin: 18px + 30px 0%;
          opacity: 0;
        }

        .subNav:focus-within {
          pointer-events: auto;
          transform: scale(1, 1);
          opacity: 1;
        }
      }
    }
  }

  @media (max-width: 767px) {
    position: relative;
    height: 70px;

    .logoContainer {
      background-color: white;
      box-shadow: 0 3px 3px 0 rgba(0, 0, 0, 0.1);

      .logoWrapper {
        width: auto;
        padding: 15px 0;
        float: left;

        &:before {
          display: none;
        }

        .logo {
          padding: 0;
          position: static;
          background-color: transparent;
          box-shadow: none;
          width: 125px;
          height: 40px;
        }
      }
    }

    .burgerMenu {
      display: inline-block;
    }

    nav {
      border-bottom-left-radius: 4px;
      background-color: rgba(255, 255, 255, 0.95);
      box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.5);
      max-height: 95%;
      max-width: 90%;
      position: fixed;
      right: 0;
      top: 0;
      padding: 70px 0 0 0;
      white-space: nowrap;
      overflow-y: scroll;
      transform: translateX(100%);
      pointer-events: none;
      opacity: 0;
      transition: all 200ms ease-in-out;

      .menuWrapper {
        margin: 0 20px;
        display: block;

        .menuItemWrapper,
        .darkModeToggle {
          display: block;
          color: black;
          padding: 10px 34px 0;
          text-align: left;

          .menuItem {
            padding: 0;
            display: block;
          }
        }

        .darkModeToggle {
          text-align: left;
          padding: 10px 34px;
          margin: 15px 0 0;

          @media (max-width: 767px) {
            text-align: center;
          }

          a {
            margin: 0;
            padding: 0;
          }
        }

        &.active {
          &:after {
            display: none;
          }
        }

        .subNav {
          padding: 0 0 0 25px;
          font-weight: 200;
          display: block;

          ul {
            margin: 0;
            padding: 0;
          }
        }
      }
    }
  }
}

.menuOpen {
  nav {
    opacity: 1;
    pointer-events: auto;
    transform: translateX(0);
  }

  .burgerMenu {
    span:nth-child(1) {
      transform: rotate(45deg);
      top: 7px;
    }

    span:nth-child(2) {
      transform: rotate(45deg);
      opacity: 0;
    }

    span:nth-child(3) {
      transform: rotate(-45deg);
      top: -7px;
    }
  }
}

.darkModeHeader {
  transition: background-color 1s ease;
  color: $dark-heading;

  .logoContainer {
    transition: background-color 1s ease;
    background-color: $dark-heading;
  }

  .navContainer {
    @media (max-width: 767px) {
      background-color: $dark-heading;
      color: $primary;

      .menuWrapper > .menuItemWrapper > .menuItem {
        color: $primary;
      }
    }
  }

  .logoWrapper:before {
    transition: background-color 1s ease;
    background-color: $dark-heading;
  }

  .logoWrapper {
    transition: background-color 1s ease;
    background-color: $dark-heading;

    .logo {
      transition: background-color 1s ease;
      background-color: $dark-heading;
    }
  }

  .menuWrapper > .menuItemWrapper > .subNav > ul {
    transition: background-color 1s ease;
    background-color: $dark-heading;

    a {
      transition: background-color 1s ease;
      color: white;
    }
  }

  .menuWrapper > .menuItemWrapper > .subNav {
    &:before,
    &:after {
      transition: background-color 1s ease;
      background-color: $dark-heading;
    }
  }

  .burgerMenu > span {
    background-color: $primary;
  }
}
