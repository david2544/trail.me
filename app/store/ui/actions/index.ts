import ActionTypes from '@store/actionTypes';
import { action } from 'typesafe-actions';

const changeTheme = (darkMode: boolean) => action(ActionTypes.CHANGE_DARK_MODE, darkMode);

// eslint-disable-next-line import/prefer-default-export
export { changeTheme };
