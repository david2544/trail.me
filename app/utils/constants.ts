export const PRODUCTS = {
  riester: 'riester',
  ruerup: 'ruerup',
  bav: 'bav',
  robo: 'robo',
  general: 'general',
};

export const FAIRR_PRODUCTS = {
  riester: 'fairriester',
  ruerup: 'fairruerup',
  bav: 'fairrbav',
  robo: 'fairrobo',
};

// WIP here. Links will probably be handled differently
export const LINKS = {
  home: '',
  cockpit: '/cockpit',
  security: '/security',
  switch: {
    general: '/change',
    riester: '/change/riester',
    ruerup: '/change/ruerup',
  },
  products: {
    riester: {
      plp: '/products/fairriester',
      calculator: '/products/fairriester/calculator',
    },
    ruerup: {
      plp: '/products/fairruerup',
      calculator: '/products/fairruerup/calculator',
    },
    bav: {
      plp: '/products/fairrbav',
    },
    robo: {
      plp: '/products/fairrobo',
    },
    raisin: {
      savings: 'https://www.weltsparen.de/tagesgeld/',
      term: 'https://www.weltsparen.de/tagesgeld/',
    },
  },
};

export const LOCALES = {
  de: 'de',
  en: 'en',
};

export const ALL = 'all';
