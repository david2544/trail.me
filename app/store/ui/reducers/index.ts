import ActionTypes from '@store/actionTypes';
import { UIActions, UIState } from '@store/ui/actions/types';

const DEFAULT_LOCALE = 'en';

export const initialState: UIState = {
  locale: DEFAULT_LOCALE,
};

export default (state: UIState = initialState, action: UIActions): UIState => {
  switch (action.type) {
    case ActionTypes.CHANGE_LOCALE:
      return {
        locale: action.payload,
      };
    default:
      return state;
  }
};
