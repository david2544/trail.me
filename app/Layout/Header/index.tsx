import React from 'react';
import { Link } from 'react-router-dom';
import { History } from 'history';
import classnames from 'classnames';
import useScroll from '@app/hooks/useScroll';
import DarkModeToggle from 'react-dark-mode-toggle';
import useToggleDarkMode from '@app/hooks/useToggleDarkMode';
import { LINKS } from '@utils/constants';
import Container from '@common/Container';
import Logo from '@images/logo.png';
import { isMenuItemActive } from '../utils';
import styles from './styles.module.scss';

interface HeaderProps {
  history: History;
  toggleMenu: Function;
  menuOpen: boolean;
}

const Header: React.FC<HeaderProps> = ({
  history: {
    location: { pathname },
  },
  toggleMenu,
  menuOpen,
}) => {
  const { isDarkMode, toggleDarkMode } = useToggleDarkMode();

  const { scrollPosition } = useScroll();

  return (
    <div
      className={classnames(styles.mainHeaderNav, {
        [styles.menuOpen]: menuOpen,
        [styles.darkModeHeader]: isDarkMode,
      })}
    >
      <div className={styles.logoContainer}>
        <Container>
          <div
            className={classnames(styles.logoWrapper, {
              [styles.minimized]: scrollPosition > 200,
            })}
          >
            <Link to={LINKS.home}>
              <img alt="fairr-logo" className={styles.logo} src={Logo} />
            </Link>
          </div>
          <button
            type="button"
            tabIndex={0}
            onClick={() => toggleMenu(!menuOpen)}
            className={styles.burgerMenu}
          >
            <span />
            <span />
            <span />
          </button>
        </Container>
      </div>

      <nav className={classnames('container', styles.navContainer)}>
        <ul className={styles.menuWrapper}>
          <li
            className={classnames(styles.menuItemWrapper, {
              [styles.active]: isMenuItemActive.isOnTransferPages(pathname),
            })}
          >
            {/* <Link to={LINKS.switch.general} className={styles.menuItem}>
              navi.main.transfer
            </Link>
            <div className={styles.subNav}>
              <ul>
                {subHeaderItems.transfer.map(item => (
                  <SubNav key={item.description} subMenuItem={item} />
                ))}
              </ul>
            </div> */}
          </li>
          <li
            className={classnames(styles.menuItemWrapper, {
              [styles.active]: isMenuItemActive.isOnSecurityPage(pathname),
            })}
          >
            {/* <Link to={LINKS.security} className={styles.menuItem}>
              about
            </Link> */}
          </li>
          <li className={styles.menuItemWrapper}>
            <div className={`${styles.menuItem} ${styles.darkModeToggle}`}>
              <DarkModeToggle onChange={toggleDarkMode} checked={isDarkMode} size={60} />
            </div>
          </li>
        </ul>
      </nav>
    </div>
  );
};

export default Header;
