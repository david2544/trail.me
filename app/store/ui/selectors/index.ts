import { ApplicationRootState } from '@store/IStore';
import { initialState } from '../reducers';

/**
 * Direct selector to the languageToggle state domain
 */
const uiState = (state: ApplicationRootState) => state.ui || initialState;

export { uiState };
