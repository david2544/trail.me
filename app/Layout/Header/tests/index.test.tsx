import React from 'react';
import * as Redux from 'react-redux';
import { shallow } from 'enzyme';
import history from '@utils/history';
import * as actions from '@store/ui/actions';
import { LINKS } from '@utils/constants';
import Header from '../index';

jest.mock('store/ui/actions');

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as {}),
  useHistory: () => ({
    push: jest.fn(),
    listen: jest.fn(),
    location: { pathname: LINKS.home },
  }),
}));

describe('<Header />', () => {
  let useSelectorSpy: jest.SpyInstance;
  let useDispatchSpy: jest.SpyInstance;

  beforeEach(() => {
    useSelectorSpy = jest.spyOn(Redux, 'useSelector');
    useDispatchSpy = jest.spyOn(Redux, 'useDispatch');
    useSelectorSpy.mockReturnValue(jest.fn());
    useDispatchSpy.mockReturnValue(jest.fn());
  });

  const props = {
    scrollPosition: 0,
    locale: 'en',
    onLocaleToggle: jest.fn(),
    history,
    hovered: '',
    setHovered: jest.fn(),
    toggleMenu: jest.fn(),
    menuOpen: false,
  };

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<Header {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call changeLocale action on "de" locale link click', () => {
    const wrapper = shallow(<Header {...props} />);
    // link for "de" locale
    const link = wrapper.find('button').at(2);
    const newLocale = 'de';

    link.simulate('click');
    expect(actions.changeLocale).toHaveBeenCalledWith(newLocale);
    expect(actions.changeLocale).toHaveBeenCalledTimes(1);
  });

  it('should call changeLocale action on "en" locale link click', () => {
    const wrapper = shallow(<Header {...props} />);
    // link for "en" locale
    const link = wrapper.find('button').at(3);
    const newLocale = 'en';

    link.simulate('click');
    expect(actions.changeLocale).toHaveBeenCalledWith(newLocale);
    expect(actions.changeLocale).toHaveBeenCalledTimes(1);
  });

  it('should change the data-lang-hover attribute to the "de" locale', () => {
    const wrapper = shallow(<Header {...props} />);
    // link for "de" locale
    const link = wrapper.find('button').at(2);

    link.simulate('mouseEnter');
    expect(wrapper.find('[data-lang-hover="de"]').exists()).toBeTruthy();
  });

  it('should change the data-lang-hover attribute to the "en" locale', () => {
    const wrapper = shallow(<Header {...props} />);
    // link for "en" locale
    const link = wrapper.find('button').at(3);

    link.simulate('mouseEnter');
    expect(wrapper.find('[data-lang-hover="en"]').exists()).toBeTruthy();
  });

  it('should change the data-lang-hover attribute to the default locale on mouseLeave for "en" link', () => {
    // mock default value from store
    useSelectorSpy.mockReturnValue({ locale: 'someDefaultValue' });
    const wrapper = shallow(<Header {...props} />);
    // link for "de" locale
    const link = wrapper.find('button').at(2);

    link.simulate('mouseLeave');
    expect(wrapper.find('[data-lang-hover="someDefaultValue"]').exists()).toBeTruthy();
  });

  it('should change the data-lang-hover attribute to the default locale on mouseLeave for "de" link', () => {
    // mock default value from store
    useSelectorSpy.mockReturnValue({ locale: 'someDefaultValue' });
    const wrapper = shallow(<Header {...props} />);
    // link for "en" locale
    const link = wrapper.find('button').at(3);

    link.simulate('mouseLeave');
    expect(wrapper.find('[data-lang-hover="someDefaultValue"]').exists()).toBeTruthy();
  });

  it('should toggle menu when clicking the button', () => {
    // mock default value from store
    const wrapper = shallow(<Header {...props} />);
    const link = wrapper.find('button').first();

    link.simulate('click');
    expect(props.toggleMenu).toHaveBeenCalledWith(true);
    expect(props.toggleMenu).toHaveBeenCalledTimes(1);
  });
});
