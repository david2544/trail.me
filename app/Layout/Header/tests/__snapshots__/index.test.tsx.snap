// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div
  className="mainHeaderNav"
>
  <div
    className="logoContainer"
  >
    <Container>
      <div
        className="logoWrapper"
      >
        <Link
          to=""
        >
          <Image
            alt="fairr-logo"
            className="logo"
            src="IMAGE_MOCK"
          />
        </Link>
      </div>
      <button
        className="burgerMenu"
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <span />
        <span />
        <span />
      </button>
    </Container>
  </div>
  <nav
    className="container navContainer"
  >
    <ul
      className="menuWrapper"
    >
      <li
        className="menuItemWrapper"
      >
        <button
          className="menuItem"
          type="button"
        >
          navi.main.products
        </button>
        <div
          className="subNav"
        >
          <ul>
            <SubNav
              key="navi.main.products.riester.description"
              subMenuItem={
                Object {
                  "description": "navi.main.products.riester.description",
                  "icon": "TODO",
                  "link": "/products/fairriester",
                  "title": "fairriester",
                }
              }
            />
            <SubNav
              key="navi.main.products.ruerup.description"
              subMenuItem={
                Object {
                  "description": "navi.main.products.ruerup.description",
                  "icon": "TODO",
                  "link": "/products/fairruerup",
                  "title": "fairruerup",
                }
              }
            />
            <SubNav
              key="navi.main.products.bav.description"
              subMenuItem={
                Object {
                  "description": "navi.main.products.bav.description",
                  "icon": "TODO",
                  "link": "/products/fairrbav",
                  "title": "fairrbav",
                }
              }
            />
            <SubNav
              key="navi.main.products.robo.description"
              subMenuItem={
                Object {
                  "description": "navi.main.products.robo.description",
                  "icon": "TODO",
                  "link": "/products/fairrobo",
                  "title": "fairrobo",
                }
              }
            />
            <SubNav
              key="navi.external.savings-account.description"
              subMenuItem={
                Object {
                  "description": "navi.external.savings-account.description",
                  "isExternal": true,
                  "link": "https://www.weltsparen.de/tagesgeld/",
                  "title": "navi.external.savings-account",
                  "titleSub": "navi.external.by-weltsparen",
                }
              }
            />
            <SubNav
              key="navi.external.term-deposit.description"
              subMenuItem={
                Object {
                  "description": "navi.external.term-deposit.description",
                  "isExternal": true,
                  "link": "https://www.weltsparen.de/tagesgeld/",
                  "title": "navi.external.term-deposit",
                  "titleSub": "navi.external.by-weltsparen",
                }
              }
            />
          </ul>
        </div>
      </li>
      <li
        className="menuItemWrapper"
      >
        <Link
          className="menuItem"
          to="/change"
        >
          navi.main.transfer
        </Link>
        <div
          className="subNav"
        >
          <ul>
            <SubNav
              key="navi.main.transfer.riester"
              subMenuItem={
                Object {
                  "description": "navi.main.transfer.riester",
                  "link": "/change/riester",
                }
              }
            />
            <SubNav
              key="navi.main.transfer.ruerup"
              subMenuItem={
                Object {
                  "description": "navi.main.transfer.ruerup",
                  "link": "/change/ruerup",
                }
              }
            />
          </ul>
        </div>
      </li>
      <li
        className="menuItemWrapper"
      >
        <Link
          className="menuItem"
          to="/security"
        >
          navi.main.security
        </Link>
      </li>
      <li
        className="loginWrapper"
      >
        <Link
          to="/cockpit"
        >
          <Button
            className="loginButton"
            isGhost={true}
            isInverted={true}
            type="button"
          >
            <span>
              Cockpit
            </span>
          </Button>
        </Link>
      </li>
      <li
        className="contactWrapper phone"
      >
        <a
          href="tel:030 - 120 822 60"
        >
          <Paragraph
            className=""
            id=""
            lead={false}
          >
            030 - 120 822 60
          </Paragraph>
          <small
            className="phoneTimes"
          >
            global.business-hours
          </small>
        </a>
      </li>
      <li
        className="contactWrapper mail"
      >
        <a
          href="mailto:support@fairr.de"
        >
          <Paragraph
            className=""
            id=""
            lead={false}
          >
            support@fairr.de
          </Paragraph>
        </a>
      </li>
      <li
        className="langSelectorWrapper"
      >
        <div
          data-lang-hover=""
        >
          <div
            className="langSelection"
          />
          <button
            data-active={false}
            data-lang="de"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            de
          </button>
          <button
            data-active={false}
            data-lang="en"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            en
          </button>
        </div>
      </li>
    </ul>
  </nav>
</div>
`;
